cmake/Sanitizers.cmake

```diff
@@ -16,7 +16,11 @@ function(
     endif()
 
     if(${ENABLE_SANITIZER_LEAK})
-      list(APPEND SANITIZERS "leak")
+      if(CMAKE_CXX_COMPILER_ID MATCHES "AppleClang")
+        message(WARNING "Leak sanitizer is not supported on Apple Clang.")
```

Comment:
Maybe this should be an error rather than a warning given that the resulting compilation will fail anyway, right?